### DISCUSSION TITLE
Site switching not always persistent and can be confusing

### DISCUSSION THREAD

**Discussion by jsunsawyer** [craftcms/cms#8959] [Ideas] [answered]
### Description
Recently, changes have been made to attempt to keep the site set to the last version a user had switched to when navigating the CP.

One outlier that I found was when switching a site from within an entry. This results in the CP using whatever site was selected upon initially viewing the entry after a user goes back to view other entries, globals, or whatever.

In my opinion, switching the site should always save the setting. If it shouldn't, the intended functionality should be clearly outlined so we can explain it to our users to avoid confusion.

Ref: https://github.com/craftcms/cms/issues/2779

### Additional info

- Craft version: 3.5.14

**Comment by mmikkel** [comment]
I think the current behaviour is fine.

The site switchers in element indexes (and global sets) are typically perceived by authors as "global" site switchers – so it makes sense that changing the site here, also changes the "globally" selected site throughout the CP.

On the other hand, the site switcher in the entry edit form is used to edit or preview a particular entry across different sites, independently of the currently selected site in the CP. The context and positioning of this site switcher is quite clearly not "global", and I don't think most authors would expect changing the site from within an entry would also change the "globally" selected site. In fact, I know that it would really mess with a lot of authors' workflow if it did.

One size doesn't necessarily fit all, though - potentially this is something that could be made configurable?

(In my opinion what _would_ be helpful in making site switchers' behaviour less confusing, would be to replace the site switchers in element indexes and global sets with an actual global site switcher, in the global CP header)

**Comment by brandonkelly** [comment, marked as answer]
This was resolved in Craft 4 (see #8920).
